plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/github/GHService.java

```diff
@@ -0,0 +1,159 @@
+package io.jenkins.tools.pluginmodernizer.core.github;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.jenkins.tools.pluginmodernizer.core.config.Config;
+import io.jenkins.tools.pluginmodernizer.core.config.Settings;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.RefAlreadyExistsException;
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.kohsuke.github.GHIssueState;
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "false positive")
+public class GHService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GHService.class);
+
+    private final Config config;
+
+    public GHService(Config config) {
+        this.config = config;
+    }
+
+    private static final String GITHUB_TOKEN = Settings.GITHUB_TOKEN;
+    private static final String FORKED_REPO_OWNER = Settings.GITHUB_USERNAME;
+    private static final String ORIGINAL_REPO_OWNER = Settings.ORGANIZATION;
+    // TODO: Change commit message and PR title based on applied recipes
+    private static final String COMMIT_MESSAGE = "Applied transformations with specified recipes";
+    private static final String PR_TITLE = "Automated PR";
+
+    public void forkCloneAndCreateBranch(String pluginName, String branchName) throws IOException, GitAPIException, InterruptedException {
+        Path pluginDirectory = Paths.get(Settings.TEST_PLUGINS_DIRECTORY, pluginName);
+
+        GitHub github = GitHub.connectUsingOAuth(GITHUB_TOKEN);
+        GHRepository originalRepo = github.getRepository(ORIGINAL_REPO_OWNER + "/" + pluginName);
+
+        getOrCreateForkedRepo(github, originalRepo);
+
+        cloneRepositoryIfNeeded(pluginDirectory, pluginName);
+
+        createAndCheckoutBranch(pluginDirectory, branchName);
+    }
+
+    private void getOrCreateForkedRepo(GitHub github, GHRepository originalRepo) throws IOException, InterruptedException {
+        GHRepository forkedRepo = github.getMyself().getRepository(originalRepo.getName());
+        if (forkedRepo == null) {
+            LOG.info("Forking the repository...");
+            originalRepo.fork();
+            Thread.sleep(5000); // Ensure the completion of Fork
+            LOG.info("Repository forked successfully.");
+        } else {
+            LOG.info("Repository already forked.");
```

Comment:
Can you please track an issue to ensure the GH repo is in sync with upstream. It should be possible via API

I'm using generally the `gh repo sync username/repo`

Without sync existing repo there is a risk to open branches from outdated main branch